src/main/java/net/journey/client/render/RenderEssenceBar.java

```diff
@@ -34,20 +34,21 @@ public void onRender(RenderGameOverlayEvent.Post event) {
 
 	private void renderManaBar(Minecraft mc, EntityPlayer player) {
 		ItemStack heldItemMainhand = player.getHeldItemMainhand();
+		IEssence mana = player.getCapability(EssenceProvider.ESSENCE_CAP, null);
 
 		if (mc.currentScreen == null && instanceOfEssenceItem(heldItemMainhand.getItem()) && transparency < 1.0) {
-			transparency += .02;
+			transparency += 0.02;
+		} else if (mana.getEssenceValue() < mana.getMaxValue()) {
```

Comment:
what's happening here?

Reply:
I mean, the whole logic of transparency

Reply:
Transparency is a float value representing the transparency of the bar from 0 to 1. If the player is holding an essence item and has less than max essence (should probably use the maxValue method now that I think about it), transparency is increased by a small amount. Otherwise if essence isn't the max value an if and else if tries to move the transparency value towards 0.36. If none of the above is true (aka no essence item is held and essence is max) it will decrease the transparency value until the bar is gone.

What this does is:
-if you're holding an essence item transparency will approach max

-if you're not holding an essence item but you don't have full essence the transparency will approach 0.36

-if you're not holding an essence item and you have full essence the transparency will approach 0